lib/ui/isolate_name_server/isolate_name_server.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/isolate_name_server/isolate_name_server.h"
+
+namespace blink {
+
+Dart_Port IsolateNameServer::LookupIsolatePortByName(const std::string& name) {
+  std::unique_lock<std::mutex> lock(mutex_);
+  return LookupIsolatePortByNameUnprotected(name);
+}
+
+Dart_Port IsolateNameServer::LookupIsolatePortByNameUnprotected(
+    const std::string& name) {
+  auto port_iterator = port_mapping_.find(name);
+  if (port_iterator != port_mapping_.end()) {
+    return port_iterator->second;
+  }
+  return ILLEGAL_PORT;
+}
+
+bool IsolateNameServer::RegisterIsolatePortWithName(Dart_Port port,
```

Comment:
Lets use the annotations in `//flutter/fml/synchronization/thread_annotations.h` for statically enforcing threading rules.

Reply:
If you define _LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS (which the default build flags already do) you get those annotations for free.

Reply:
Ah okay. Turns out I wasn't doing things totally right and was getting warnings as a result. Done.